pipeline/src/pipeline.rs

```diff
@@ -114,6 +116,10 @@ pub struct Pipeline<T: FieldElement> {
     artifact: Artifacts<T>,
     /// Output directory for intermediate files. If None, no files are written.
     output_dir: Option<PathBuf>,
+    /// The temporary directory, owned by the pipeline (or any copies of it).
+    /// This object is not used directly, but keeping it here ensures that the directory
+    /// is not deleted until the pipeline is dropped.
+    _tmp_dir: Option<Rc<Temp>>,
```

Comment:
This is much more convenient and save, otherwise you have to keep around the `Temp` object around for as long as the pipeline lives.

I think we used to do it like that before, not sure we stopped? Maybe because we wanted `Pipeline` to be clone-able? But the `Rc` does exactly what you'd want.